### ISSUE TITLE
Can this be used?

### ISSUE THREAD

**Issue by Johan-dutoit** [Polidea/react-native-blemulator#90]
Hi,

Firstly, thanks for the great library ble-plx, it's been of tremendous use.

As per the title, is the emulator ready to be used?  What's the state of it, etc?
If not, are there any areas that I can help contribute towards?

Cheers

**Comment by mikolak** [comment]
Hi!

react-native-blemulator is feature complete on Android and can be used there, but iOS is barely started and any attempt at using it will result in exceptions thrown (best case scenario) or nothing at all (worst case scenario).

I'm tracking work here: https://github.com/Polidea/react-native-blemulator/projects/1
but since I was the only one working on it, they are barely described at all. iOS part has to be written in Objective-C to avoid any unnecessary issues with Swift compatibility (we have plans to migrate MultiPlatform BLE Adapter to pure obj-c as well). There are two references points: Android implementation and Flutter's BLEmulator (https://github.com/Polidea/blemulator_flutter/tree/master/ios/Classes)

As you can see I have a WIP MR to set up the communication between JavaScript and Obj-C, but I didn't have to time to get back to it since August. All help would be appreciated. 

Communication is a blocking feature at the moment, nothing else can be implemented until it's up and running.

Cheers

**tl;dr** you're good to go on Android, it won't even start on iOS.

**Comment by orome** [comment]
Is this still nonfunctional on iOS?
